src/main/java/io/appform/dropwizard/actors/base/UnmanagedConsumer.java

```diff
@@ -90,6 +92,12 @@ private Handler(Channel channel,
         public void handleDelivery(String consumerTag, Envelope envelope,
                                    AMQP.BasicProperties properties, byte[] body) throws IOException {
             try {
+                if (properties.getHeaders().containsKey("X-CompressionType")) {
+                    log.info("Compressed message length: {}", body.length);
+                    body = CompressionProvider.decompress(body, (CompressionAlgorithm) properties.getHeaders()
+                            .get("X-CompressionType"));
+                    log.info("Decompressed message length: {}", body.length);
```

Comment:
Introduce a BodyDecorator. It is compression now, it can be encryption later. This will lay down the groundwork for the same